### PR TITLE
fabric8-launcher: make Launcher Booster Catalog customizable

### PR DESCRIPTION
**Pull request by vorburger**
Useful e.g. for @edewit's https://github.com/edewit/minecraft-sponge-plugin-s2i/issues/4, but also for anyone else who wants to use the fabric8-launcher addon but with a custom Booster Catalog. 

There is a comment at the top of https://raw.githubusercontent.com/fabric8-launcher/launcher-documentation/master/docs/topics/proc_installing-launcher-tool-addon.adoc for something similar? Perhaps this PR should be pointed out to someone working on that - @tradej @inoxx03 @rhoads-zach ? BTW: I don't understand how that documentation can possibly be a "parametrized module", given that parameter-custom-catalog doesn't actually appear in that doc?

It's Not Great at all that I had to copy/paste and repeat the default values for the ..REPOSITORY and ..REF from the original [launcher-template.yaml](https://raw.githubusercontent.com/fabric8-launcher/launcher-openshift-templates/master/openshift/launcher-template.yaml) here - anyone knows if there is a better way?

I haven't actually tested this, yet.

### CODE REVIEW

**Comment by praveenkumar** [comment]
@vorburger Can someone from your team or you test this change because we are not so expert about what to test here and how it should behave.

**Comment by vorburger** [comment]
@praveenkumar I'll test it, but do you have any idea / know anyone who could help re. a better way to avoid the default values for the ..REPOSITORY and ..REF from the original launcher-template.yaml here?

**Comment by praveenkumar** [comment]
@vorburger I am not sure, might be good to ask on launcher github issue tracker?

**Comment by vorburger** [comment]
@praveenkumar but it isn't really a Launcher specific issue... this is more of a minishift-addon thing: So I want users to have the option to override those two things, but they are options which already have default values, in the YAML template. So if a user does not specify those 2 options, those should be used. But I have copy/pasted those from the YAML template into the minishift addon, and in an ideal world that should be avoided I guess. Do you see what I mean?

**Comment by praveenkumar** [comment]
@vorburger let me know if I understand correctly, as of now you put those value as part of `Default-Vars` but you also want this value can be over ride by user then also put those variable as part of `Required-Vars` so that user can override using like below.

```
$ minishift addon apply -a "LAUNCHER_BOOSTER_CATALOG_REPOSITORY=foo.com" -a "LAUNCHER_BOOSTER_CATALOG_REF=bar"
```

File: add-ons/fabric8-launcher/fabric8-launcher.addon [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # Description: Installs Fabric8 Launcher in a project named rhoarpad 
 # Required-Vars: GITHUB_USERNAME, GITHUB_TOKEN,OPENSHIFT_USERNAME,OPENSHIFT_PASSWORD 
 # OpenShift-Version: >=3.9.0 
-# Var-Defaults: OPENSHIFT_USERNAME=developer,OPENSHIFT_PASSWORD=developer,CONSOLE_URL=https://#{ip}:8443,LAUNCHER_PROJ=rhoarpad
+# Var-Defaults: OPENSHIFT_USERNAME=developer,OPENSHIFT_PASSWORD=developer,CONSOLE_URL=https://#{ip}:8443,LAUNCHER_PROJ=rhoarpad,LAUNCHERLAUNCHER_BOOSTER_CATALOG_REPOSITORY=https://github.com/fabric8-launcher/launcher-booster-catalog.git,LAUNCHER_BOOSTER_CATALOG_REF=latest
```
Comment (praveenkumar) [inline add-ons/fabric8-launcher/fabric8-launcher.addon]: Typo `LAUNCHERLAUNCHER_BOOSTER_CATALOG_REPOSITORY` should be `LAUNCHER_BOOSTER_CATALOG_REPOSITORY`

File: add-ons/fabric8-launcher/fabric8-launcher.addon [hunk 2]
```diff
@@ -17,4 +17,4 @@ oc new-project #{LAUNCHER_PROJ}
...
 
 # instantiate launcher
-oc new-app fabric8-launcher --name=launcher -n #{LAUNCHER_PROJ} -p LAUNCHER_MISSIONCONTROL_GITHUB_USERNAME=#{GITHUB_USERNAME} -p LAUNCHER_MISSIONCONTROL_GITHUB_TOKEN=#{GITHUB_TOKEN} -p LAUNCHER_MISSIONCONTROL_OPENSHIFT_CONSOLE_URL=#{CONSOLE_URL} -p LAUNCHER_MISSIONCONTROL_OPENSHIFT_API_URL=#{CONSOLE_URL} -p LAUNCHER_MISSIONCONTROL_OPENSHIFT_USERNAME=#{OPENSHIFT_USERNAME} -p LAUNCHER_MISSIONCONTROL_OPENSHIFT_PASSWORD=#{OPENSHIFT_PASSWORD} -p LAUNCHER_KEYCLOAK_URL= -p LAUNCHER_KEYCLOAK_REALM=
+oc new-app fabric8-launcher --name=launcher -n #{LAUNCHER_PROJ} -p LAUNCHER_MISSIONCONTROL_GITHUB_USERNAME=#{GITHUB_USERNAME} -p LAUNCHER_MISSIONCONTROL_GITHUB_TOKEN=#{GITHUB_TOKEN} -p LAUNCHER_MISSIONCONTROL_OPENSHIFT_CONSOLE_URL=#{CONSOLE_URL} -p LAUNCHER_MISSIONCONTROL_OPENSHIFT_API_URL=#{CONSOLE_URL} -p LAUNCHER_MISSIONCONTROL_OPENSHIFT_USERNAME=#{OPENSHIFT_USERNAME} -p LAUNCHER_MISSIONCONTROL_OPENSHIFT_PASSWORD=#{OPENSHIFT_PASSWORD} -p LAUNCHER_KEYCLOAK_URL= -p LAUNCHER_KEYCLOAK_REALM= -p LAUNCHER_BOOSTER_CATALOG_REPOSITORY=#{LAUNCHERLAUNCHER_BOOSTER_CATALOG_REPOSITORY} -p LAUNCHER_BOOSTER_CATALOG_REF=#{LAUNCHER_BOOSTER_CATALOG_REF}
```
Comment (praveenkumar) [inline add-ons/fabric8-launcher/fabric8-launcher.addon]: ditto here.

**Review by praveenkumar — CHANGES_REQUESTED** [review]

**Comment by vorburger** [comment]
Hello @praveenkumar I'm doing a year end clean up of my personal https://github.com/pulls and wanted to ask if you would still consider merging this very old Pull Request if I rebased it, or if we should forget about and close this? (This is a bulk message I'm posting to many issues to gauge what is still revelant.)

**Comment by gbraad** [comment]
We do not support minishift anymore. The tool was made EOL at the end of
August.

No member of the team has time to spend on testing or rebasing. If you do
this, perhaps it can be merged.

**Pull request closed without merging**